sre-recipes/recipes/recipe_1/recipe_1.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This module contains the implementation of recipe 1
+"""
+
+import os
+import subprocess
+from recipe import Recipe
+
+
+class Recipe1(Recipe):
```

Comment:
Why is this Recipe1 and extending the Recipe base class?  Is this to support the first approach to SRE recipes?  With the broken and fixed builds being swapped and "Recipe2" would be another approach?

If so I think we need a more descriptive class name than Recipe1.  Also I might consider making this an ABC or just a class that is invoked with a different service param each time to vary the service used.

Reply:
Yeah, Recipes is an ABC that defines methods/behavior that each recipe must have. Any recipe would need to extend ABC in order to be considered a recipe. The goal is that each recipe is contained within a folder so that we have as little to maintain as possible. It also makes it easier for people adding recipes to have flexibility over the implementation of breaking and restoring services. Simon's goal is to have the implementation of breaking and restoring be the responsibility of the author of the recipe instead of the CLI tool. This means that we probably don't want to make Recipe1 an ABC that is invoked with a different service param because we don't want that as our responsibility.

As for the naming, I can rename the class to something like ConvertCurrencies instead.